src/main/ruby/core/stat.rb

```diff
@@ -73,29 +72,30 @@ class StatStruct < Rubinius::FFI::Struct
     S_ISVTX  = Rubinius::Config['rbx.platform.file.S_ISVTX']
 
     attr_reader :path
-    
-    attr_reader :struct
-    
-    def self.__allocate__
-      instance = super
-      instance.instance_variable_set :@struct, StatStruct.new
-      instance
-    end
 
-    def initialize(path)
-      path = Rubinius::Type.coerce_to_path(path)
-      result = Truffle::POSIX.stat(path, struct)
-      Errno.handle_nfi path unless result == 0
+    def initialize(path_or_buffer)
+      if path_or_buffer.is_a?(Buffer)
+        @buffer = path_or_buffer.to_h
+      else
+        path = Rubinius::Type.coerce_to_path(path_or_buffer)
+        Buffer.new { |buffer|
```

Comment:
It's more idiomatic to use `do` ... `end` instead of curly braces for blocks that span multiple lines.

Reply:
I'm using `{ }` because I'm using the return value - so it's like an expression rather than a statement. Is that not common?

Reply:
I've never heard about this distinction in this context, I think both are considered expressions and the convention is to use `do`...`end` because it's multiline. An explicit return could be used to highlight the whole thing is the return value.

Reply:
Fixed.